collect_app/src/main/java/org/odk/collect/android/tasks/MediaLoadingTask.java

```diff
@@ -56,7 +56,7 @@ protected File doInBackground(Uri... uris) {
                 String destMediaPath = instanceFolder + File.separator + System.currentTimeMillis() + extension;
 
                 try {
-                    File chosenFile = MediaUtils.getFileFromUri(formEntryActivity.get(), uris[0], MediaStore.Images.Media.DATA);
+                    File chosenFile = MediaUtils.getFileFromUri(formEntryActivity.get(), uris[0], MediaStore.Images.Media.DATA, formEntryActivity.get().getNetworkStateProvider());
```

Comment:
I think we could pass the `NetworkStateProvider` in here as a constructor arg rather than needing to have the getter (which I know understand) on `FormEntryActivity`.

Reply:
Done.